src/main/mw.c

```diff
@@ -173,18 +173,37 @@ bool isCalibrating()
     return (!isAccelerationCalibrationComplete() && sensors(SENSOR_ACC)) || (!isGyroCalibrationComplete());
 }
 
+#define RC_RATE_INCREMENTAL 0.13545f
+
 float calculateSetpointRate(int axis, int16_t rc) {
-    float angleRate;
+    float angleRate, rcMax, rcRate, rcSuperfactor;
+    uint8_t rcExpo;
+
+    if (axis != YAW) {
+        rcExpo = currentControlRateProfile->rcExpo8;
+        rcRate = (currentControlRateProfile->rcRate8 <= 200) ? currentControlRateProfile->rcRate8 : currentControlRateProfile->rcRate8 + (RC_RATE_INCREMENTAL * (currentControlRateProfile->rcRate8 - 200));
```

Comment:
I think there is some sort of scaling problem in here.

For `currentControlRateProfile->rcRate8` = 255 I get

`rc_rate = 255 + (0.13545f * (255 - 200))`

`rc_rate = 255 + 7.44975 = 262.44975`